tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestOperations/UnifiedAggregateCollectionOperation.cs

```diff
@@ -0,0 +1,108 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format.UnifiedTestOperations
+{
+    public class UnifiedAggregateCollectionOperation : IUnifiedTestOperation
+    {
+        private IMongoCollection<BsonDocument> _collection;
+        private AggregateOptions _options;
+        private PipelineDefinition<BsonDocument, BsonDocument> _pipeline;
+
+        public UnifiedAggregateCollectionOperation(
+            IMongoCollection<BsonDocument> collection,
+            AggregateOptions options,
+            PipelineDefinition<BsonDocument, BsonDocument> pipeline)
+        {
+            _collection = collection;
+            _options = options;
+            _pipeline = pipeline;
+        }
+
+        public OperationResult Execute(CancellationToken cancellationToken)
+        {
+            List<BsonDocument> result;
+
+            try
+            {
+                var cursor = _collection.Aggregate(_pipeline, _options, cancellationToken);
+                result = cursor.ToList();
+            }
+            catch (Exception ex)
+            {
+                return new OperationResult(ex);
+            }
+
+            return new OperationResult(new BsonArray(result));
+        }
+
+        public async Task<OperationResult> ExecuteAsync(CancellationToken cancellationToken)
+        {
+            List<BsonDocument> result;
+
+            try
+            {
+                var cursor = await _collection.AggregateAsync(_pipeline, _options, cancellationToken);
+                result = cursor.ToList();
+            }
+            catch (Exception ex)
+            {
+                return new OperationResult(ex);
+            }
+
+            return new OperationResult(new BsonArray(result));
+        }
+    }
+
+    public class UnifiedAggregateCollectionOperationBuilder
```

Comment:
And because of the comment above there should be some intermediary between operation description from test document and `IUnifiedTestOperation`, which parses bson arguments and converts them into operation dependencies.

Reply:
This is good.

If we ever need to support multiple unified test format schemas at the same time we might have a build for each schema version (or the builder could simply support multiple schemas at the same time with conditional statements).